test/e2e/create_test.go

```diff
@@ -169,25 +215,44 @@ func (tc *testContext) waitForWindowsNodes(nodeCount int32, waitForAnnotations,
 	return err
 }
 
+// generatePrivateKey generates a random RSA private key
+func generatePrivateKey() ([]byte, error) {
+	var keyData []byte
+	key, err := rsa.GenerateKey(rand.Reader, 1024)
+	if err != nil {
+		return nil, errors.Wrap(err, "error generating key")
+	}
+	var privateKey = &pem.Block{
+		Type:  "RSA PRIVATE KEY",
+		Bytes: x509.MarshalPKCS1PrivateKey(key),
+	}
+	buf := bytes.NewBuffer(keyData)
+	err = pem.Encode(buf, privateKey)
+	if err != nil {
+		return nil, errors.Wrap(err, "error encoding generated private key")
+	}
+	return buf.Bytes(), nil
+}
+
 // createPrivateKeySecret ensures that a private key secret exists with the correct data
-func (tc *testContext) createPrivateKeySecret() error {
-	secretsClient := tc.kubeclient.CoreV1().Secrets(tc.namespace)
-	if _, err := secretsClient.Get(context.TODO(), secrets.PrivateKeySecret, metav1.GetOptions{}); err != nil {
-		if !apierrors.IsNotFound(err) {
-			return errors.Wrap(err, "could not get private key secret")
+func (tc *testContext) createPrivateKeySecret(useKnownKey bool) error {
+	if err := tc.ensurePrivateKeyDeleted(); err != nil {
+		return errors.Wrap(err, "error ensuring any existing private key is removed")
+	}
+	var keyData []byte
+	var err error
+	if useKnownKey {
+		keyData, err = ioutil.ReadFile(gc.privateKeyPath)
+		if err != nil {
+			return errors.Wrapf(err, "unable to read private key data from file %s", gc.privateKeyPath)
 		}
 	} else {
-		// Secret already exists, delete it
-		if err := secretsClient.Delete(context.TODO(), secrets.PrivateKeySecret, metav1.DeleteOptions{}); err != nil {
-			return errors.Wrap(err, "unable to delete existing private key secret")
+		keyData, err = generatePrivateKey()
+		if err != nil {
+			return errors.Wrap(err, "error generating private key")
```

Comment:
error generating private key secret*
maybe we should mention it everywhere as private key secret, WDYT?

Reply:
Thats actually just returning a private key, not a secret